tests/docker/util.sh

```
@@ -146,6 +157,73 @@ function check_env() {
+    echo "Error: ${name} is not executable: ${bin_path}" >&2
+    exit 1
+  fi
+}
+
+function append_local_binary_overrides() {
+  local -n compose_files=$1
+  local override_dir="${2:-../docker/next-gen-yaml/override}"
+  local validate_binaries="${3:-true}"
+
+  if [[ -n "${LOCAL_PD_BIN_DIR:-}" ]]; then
+    export LOCAL_PD_BIN_DIR="$(realpath "${LOCAL_PD_BIN_DIR}")"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)